python/demo/gsoc2022/gmsh_helpers.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (C) 2020 Jørgen S. Dokken
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+#
+# =========================================
+# GMSH model to dolfinx.Mesh converter
+# =========================================
+
+import numpy
+import gmsh
+
+from mpi4py import MPI
+from dolfinx.io import extract_gmsh_geometry, extract_gmsh_topology_and_markers, ufl_mesh_from_gmsh
+from dolfinx.cpp.io import perm_gmsh, distribute_entity_data
+from dolfinx.cpp.mesh import to_type, cell_entity_type
+from dolfinx.cpp.graph import AdjacencyList_int32
+from dolfinx.mesh import meshtags_from_entities, create_mesh
+
+
+def read_from_msh(filename: str, cell_data=False, facet_data=False, gdim=None):
+    """
+    Reads a mesh from a msh-file and returns the dolfin-x mesh.
+    Input:
+        filename: Name of msh file
+        cell_data: Boolean, True of a mesh tag for cell data should be returned
+                   (Default: False)
+        facet_data: Boolean, True if a mesh tag for facet data should be
+                    returned (Default: False)
+        gdim: Geometrical dimension of problem (Default: 3)
+    """
+    if MPI.COMM_WORLD.rank == 0:
+        # Check if gmsh is already initialized
+        try:
+            current_model = gmsh.model.getCurrent()
+        except ValueError:
+            current_model = None
+            gmsh.initialize()
+
+        gmsh.model.add("Mesh from file")
+        gmsh.merge(filename)
+    output = gmsh_model_to_mesh(gmsh.model, cell_data=cell_data,
+                                facet_data=facet_data, gdim=gdim)
+    if MPI.COMM_WORLD.rank == 0:
+        if current_model is None:
+            gmsh.finalize()
+        else:
+            gmsh.model.setCurrent(current_model)
+    return output
+
+
+def gmsh_model_to_mesh(model, cell_data=False, facet_data=False, gdim=None):
```

Comment:
@jhale  Should we try to get this into DOLFINx as an alternative way of reading in GMSH models directly?